src/Storages/System/StorageSystemZooKeeper.cpp

```diff
@@ -28,6 +31,128 @@ namespace ErrorCodes
     extern const int BAD_ARGUMENTS;
 }
 
+// ZkNodeCache is a trie tree to cache all the zookeeper writes. The purpose of this struct is to avoid creating/setting nodes
```

Comment:
Actually it's not "real" trie, because "real" trie should avoid creating intermediate nodes like this:
```
     A
    /  \
   B    X
  /
 C
 /
D
```

and should look like 
```
     A
   /   \
BCD      X
```
But "real" trie would be an overkill, so it's fine :)
And sorry for being bore :)

Reply:
Isn't this actually a radix tree (https://en.wikipedia.org/wiki/Radix_tree), a special case of trie? 😄